pkg/server/controller.go

```diff
@@ -129,6 +207,65 @@ func (ctrl *Controller) drainMiddleware(next http.HandlerFunc) http.HandlerFunc
 	}
 }
 
+func (ctrl *Controller) authMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		jwtCookie, err := r.Cookie(jwtCookieName)
+		if err != nil {
+			ctrl.httpServer.ErrorLog.Printf("There seems to be problem with jwt token cookie: %v", err)
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		if jwtCookie == nil {
+			ctrl.httpServer.ErrorLog.Printf("Missing jwt cookie")
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		jwtToken := jwtCookie.Value
+		token, err := jwt.Parse(jwtToken, func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+			}
+			return []byte(ctrl.config.JWTSecret), nil
+		})
+
+		if err != nil {
+			ctrl.httpServer.ErrorLog.Printf("Error parsing jwt token: %v", err)
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		claims, ok := token.Claims.(jwt.MapClaims)
+		if !ok || !token.Valid {
+			ctrl.httpServer.ErrorLog.Printf("Token not valid")
+			http.Redirect(w, r, "/login", http.StatusTemporaryRedirect)
+			return
+		}
+
+		if exp, ok := claims["exp"].(float64); ok && int64(exp) < time.Now().Unix() {
```

Comment:
Why do we need to do this on our own? `token.Claims.Valid()` which is called in `jwt.Parse` does validate expiration as well.

Reply:
you're right, I did this with a custom field first to test something and then just went along and wrote for exp for some reason as well. will remove it together with the above.